packages/tailwind/src/tailwind.tsx

```diff
@@ -1,34 +1,115 @@
 import * as React from 'react';
 import { renderToStaticMarkup } from 'react-dom/server';
-import htmlParser, { attributesToProps, domToReact, Element } from 'html-react-parser';
 import { tailwindToCSS, TailwindConfig } from 'tw-to-css';
 
-import { cleanCss, makeCssMap, getMediaQueryCss } from './utils';
+import { cssToJsxStyle } from './utils';
 
 export interface TailwindProps {
   children: React.ReactNode;
   config?: TailwindConfig;
 }
 
-export const Tailwind: React.FC<TailwindProps> = ({ children, config }) => {
+function processElement(
+  element: React.ReactElement,
+  headStyles: string[],
+  config?: TailwindConfig
+): React.ReactElement {
   const { twi } = tailwindToCSS({
     config
   });
 
-  const newChildren = React.Children.toArray(children);
+  if (element.props.className) {
+    const convertedStyles: string[] = [];
+    const responsiveStyles: string[] = [];
+    const classNames = element.props.className.split(' ');
+
+    const customClassNames = classNames.filter((className: string) => {
+      if (twi(className, { ignoreMediaQueries: true })) {
+        convertedStyles.push(twi(className, { ignoreMediaQueries: true }));
+        return false;
+      } else if (twi(className, { ignoreMediaQueries: false })) {
+        responsiveStyles.push(className);
+        return false;
+      }
+      return true;
+    });
+
+    const convertedResponsiveStyles = twi(responsiveStyles, {
+      ignoreMediaQueries: false,
+      merge: false
+    });
+
+    headStyles.push(convertedResponsiveStyles.replace(/^\n+/, '').replace(/\n+$/, ''));
+
+    const newProps = {
+      ...element.props,
+      // eslint-disable-next-line no-undefined
+      className: customClassNames.length ? customClassNames.join(' ') : undefined,
+      style: { ...element.props.style, ...cssToJsxStyle(convertedStyles.join(' ')) }
+    };
+
+    // eslint-disable-next-line no-param-reassign
+    element = React.cloneElement(element, newProps);
+  }
+
+  if (element.props.children) {
+    const children = React.Children.toArray(element.props.children);
+    const processedChildren = children.map((child) => {
+      if (React.isValidElement(child)) {
+        return processElement(child, headStyles, config);
+      }
+      return child;
+    });
+
+    // eslint-disable-next-line no-param-reassign
+    element = React.cloneElement(element, element.props, ...processedChildren);
+  }
+
+  return element;
+}
 
-  const fullHTML = renderToStaticMarkup(<>{newChildren}</>);
+function processHead(child: React.ReactElement, responsiveStyles: string[]): React.ReactElement {
+  // FIXME: find a cleaner solution for child as any
```

Comment:
@shellscape do you know a cleaner solution for this check?

Reply:
maybe https://stackoverflow.com/a/65794258/416845

Reply:
This doesn't look cleaner, as unknown is just a longer as any